frontend/src/container/QueryBuilder/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { QueryBuilderStateT } from 'constants/queryBuilder';
+import React, {
+	createContext,
+	useCallback,
+	useContext,
+	useMemo,
+	useState,
+} from 'react';
+
+interface QueryBuilderContainerPropsI {
+	children: React.ReactElement | React.ReactNode;
+}
+
+interface InitialStateI {
+	search: string;
+}
+
+type VoidFunction = (value: string) => void;
+
+interface ContextValueI {
+	values: InitialStateI;
+	onChangeHandler: (type: QueryBuilderStateT) => VoidFunction;
+	onSubmitHandler: () => void;
+}
+
+const initialState: InitialStateI = {
+	search: '',
+};
+
+const defaultContextValue: ContextValueI = {
+	values: initialState,
+	onChangeHandler: (): VoidFunction => (): void => {},
+	onSubmitHandler: (): void => {},
```

Comment:
```suggestion
	onSubmitHandler: VoidFunction,
```

Reply:
it is not type or interface . This is initial value. We cannot use type as function here.

Reply:
make sense